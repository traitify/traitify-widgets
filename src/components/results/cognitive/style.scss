@import "style/helpers";

.container {
  @extend %container;

  text-align: center;

  a, a:visited {
    text-decoration: none;

    @include theme("color", "theme-color");
  }
  img {
    margin: 0 auto;
    max-width: 200px;
    width: 50%;
  }
}
.heading {
  @extend %heading;

  @include theme("color", "theme-color");
}
