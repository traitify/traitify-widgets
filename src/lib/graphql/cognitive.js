export const create = `
  mutation($localeKey: String, $surveyID: String!) {
    createCognitiveTest(localeKey: $localeKey, surveyId: $surveyID) {
      allottedTime
      completed
      id
      isSkipped
      learningDisability
      localeKey
      name
      percentileScore
      profileId
      rawScore
      specialAllottedTime
      stenScore
      surveyId
      surveyKey
      questions {
        id
        answer {
          answerId
          skipped
          timeTaken
        }
        questionImage {
          id
          url
        }
        responses {
          id
          image {
            id
            url
          }
        }
      }
    }
  }
`;

export const get = `
  query cognitiveTest($localeKey: String, $testID: String!) {
    cognitiveTest(localeKey: $localeKey, testId: $testID) {
      allottedTime
      completed
      id
      isSkipped
      learningDisability
      localeKey
      name
      percentileScore
      profileId
      rawScore
      specialAllottedTime
      stenScore
      surveyId
      surveyKey
      questions {
        id
        answer {
          answerId
          skipped
          timeTaken
        }
        questionImage {
          id
          url
        }
        responses {
          id
          image {
            id
            url
          }
        }
      }
    }
  }
`;

export const skip = `
  mutation($testID: String!) {
    skipCognitiveTest(testId: $testID) {
      message
      success
    }
  }
`;

export const surveys = `
  query($first: Int = 50) {
    cognitiveSurveys(first: $first) {
      edges {
        node {
          id
          key
          name
        }
      }
    }
  }
`;

export const update = `
  mutation(
    $answers: [QuestionAnswer]!,
    $learningDisability: Boolean,
    $overallTimeTaken: Int!,
    $testID: String!
  ) {
    completeCognitiveTest(
      answers: $answers,
      learningDisability: $learningDisability,
      overallTimeTaken: $overallTimeTaken,
      testId: $testID
    ) {
      message
      success
    }
  }
`;

export const path = "/cognitive-tests/graphql";
