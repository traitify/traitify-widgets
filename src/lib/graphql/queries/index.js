export benchmarks from "./benchmarks";
export benchmark from "./benchmark";
export * as cognitive from "./cognitive";
export guide from "./guide";
