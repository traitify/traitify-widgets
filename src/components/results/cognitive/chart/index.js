import times from "lib/common/array/times";
import useComponentEvents from "lib/hooks/use-component-events";
import useDisabledComponent from "lib/hooks/use-disabled-component";
import useOption from "lib/hooks/use-option";
import useResults from "lib/hooks/use-results";
import useSetting from "lib/hooks/use-setting";
import useTranslate from "lib/hooks/use-translate";
import style from "./style.scss";

export default function CognitiveChart() {
  const disabled = useDisabledComponent("CognitiveChart");
  const results = useResults({surveyType: "cognitive"});
  const allowed = useSetting("showCognitiveResults", {fallback: false});
  const showHeaders = useOption("showHeaders");
  const translate = useTranslate();

  useComponentEvents("CognitiveChart");

  if(!allowed) { return null; }
  if(disabled) { return null; }
  if(!results) { return null; }
  if(!results.stenScore) { return null; }

  const data = times(10).map((index) => index + 1).map((index) => ({
    active: index === results.stenScore,
    label: index,
    rank: index > 4 ? "high" : "low"
  }));

  return (
    <div className={style.container}>
      {showHeaders && (
        <div className={style.sectionHeading}>{translate("results.cognitive.breakdown.heading")}</div>
      )}
      <div className={style.p}>{translate("results.cognitive.breakdown.description", {survey_name: results.name})}</div>
      <div className={style.chart}>
        <div className={style.grid}>
          <div className={style.row}>
            {data.map(({active, label, rank}) => (
              <div
                key={label}
                className={[active && style.active, style[rank]].filter(Boolean).join(" ")}
              />
            ))}
          </div>
        </div>
        <div className={style.scale}>
          {data.map(({label}) => <div key={label}>{label}</div>)}
        </div>
      </div>
    </div>
  );
}
