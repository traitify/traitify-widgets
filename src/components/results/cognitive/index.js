import DangerousHTML from "components/common/dangerous-html";
import useTranslate from "lib/hooks/use-translate";
import style from "./style.scss";

export default function Cognitive() {
  const translate = useTranslate();

  return (
    <section className={style.container}>
      <img alt="Brain" src="https://cdn.traitify.com/images/cognitive/brain.png" />
      <div className={style.heading}>{translate("cognitive_results_heading")}</div>
      <DangerousHTML html={translate("cognitive_results_html")} />
    </section>
  );
}
