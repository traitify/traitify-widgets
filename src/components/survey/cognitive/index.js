/* eslint-disable no-alert */
import {useEffect, useRef, useState} from "react";
import {useRecoilRefresher_UNSTABLE as useRecoilRefresher} from "recoil";
import Loading from "components/common/loading";
import useAssessment from "lib/hooks/use-assessment";
import useCache from "lib/hooks/use-cache";
import useCacheKey from "lib/hooks/use-cache-key";
import useComponentEvents from "lib/hooks/use-component-events";
import useDidUpdate from "lib/hooks/use-did-update";
import useGraphql from "lib/hooks/use-graphql";
import useHttp from "lib/hooks/use-http";
import useOption from "lib/hooks/use-option";
import useTranslate from "lib/hooks/use-translate";
import {cognitiveAssessmentQuery} from "lib/recoil";
import Instructions from "./instructions";
import Slide from "./slide";
import Timer from "./timer";
import {useQuestionsLoader} from "./helpers";
import style from "./style.scss";

export default function Cognitive() {
  const assessment = useAssessment("cognitive");
  const assessmentCacheKey = useCacheKey("assessment");
  const cache = useCache();
  const cacheKey = useCacheKey({scope: ["slides"], type: "assessment"});
  const graphQL = useGraphql();
  const http = useHttp();
  const options = useOption("survey") || {};
  const refreshAssessment = useRecoilRefresher(cognitiveAssessmentQuery);
  const translate = useTranslate();

  const [initialQuestions, setInitialQuestions] = useState([]);
  const {dispatch, questions} = useQuestionsLoader(initialQuestions);
  const [disability, setDisability] = useState(false);
  const [onlySkipped, setOnlySkipped] = useState(false);
  const [questionIndex, setQuestionIndex] = useState(null);
  const [skipped, setSkipped] = useState(null);
  const [startTime, setStartTime] = useState(null);
  const [submitAttempts, setSubmitAttempts] = useState(0);
  const submitting = useRef(false);
  const state = {
    disability,
    dispatch,
    initialQuestions,
    questions,
    onlySkipped,
    questionIndex,
    skipped,
    startTime
  };
  const onSelect = (answer) => {
    dispatch({answer, questionIndex, type: "response"});
    if(!onlySkipped) { return setQuestionIndex(questionIndex + 1); }

    const newQuestionIndex = questions
      .findIndex((question, index) => index > questionIndex && !question.answer.answerId);
    setQuestionIndex(newQuestionIndex === -1 ? questions.length + 1 : newQuestionIndex);
  };
  const onStart = (_options) => {
    if(_options.disability) { setDisability(true); }

    setQuestionIndex(0);
    setStartTime(Date.now());
  };
  const onSubmit = () => {
    if(assessment.completed) { return; }
    if(submitAttempts > 3) { return; }
    if(submitting.current) { return; }

    submitting.current = true;

    const query = graphQL.cognitive.update;
    const variables = {
      answers: questions.map((question) => (
        question.answer ? ({
          answerId: question.answer.answerId,
          questionId: question.id,
          skipped: question.answer.skipped || false,
          timeTaken: question.answer.timeTaken < 1 ? 1 : question.answer.timeTaken
        }) : ({
          questionId: question.id,
          skipped: true, // Currently required by API
          timeTaken: 1 // Currently required by API
        })
      )),
      learningDisability: disability,
      overallTimeTaken: startTime ? parseInt((Date.now() - startTime) / 1000, 10) : 1000,
      testID: assessment.id
    };

    http.post(graphQL.cognitive.path, {query, variables}).then(({data, errors}) => {
      if(!errors && data.completeCognitiveTest.success) {
        cache.set(assessmentCacheKey, {...assessment, completed: true});
        refreshAssessment();

        submitting.current = false;
      } else {
        console.warn(errors || data); // eslint-disable-line no-console

        submitting.current = false;

        setTimeout(() => setSubmitAttempts((x) => x + 1), 2000);
      }
    });
  };

  useComponentEvents("Survey", {...state});
  useDidUpdate(() => { onSubmit(); }, [submitAttempts]);
  useEffect(() => {
    if(!assessment) { return; }

    const cachedData = cache.get(cacheKey);
    if(cachedData) {
      setDisability(cachedData.disability);
      setOnlySkipped(cachedData.onlySkipped);
      setSkipped(cachedData.skipped);

      setQuestionIndex(cachedData.questionIndex);
      setInitialQuestions(cachedData.questions);
      setStartTime(cachedData.startTime);
    } else {
      setInitialQuestions(assessment.questions);
    }
  }, [
    assessment && assessment.questions,
    assessment && assessment.localeKey
  ]);

  useEffect(() => {
    if(!assessment) { return; }
    if(questions.length === 0) { return; }

    cache.set(cacheKey, {
      disability,
      onlySkipped,
      questions,
      questionIndex,
      skipped,
      startTime
    });
  }, [disability, onlySkipped, questions, questionIndex, skipped, startTime]);

  useEffect(() => {
    if(questions.length === 0) { return; }
    if(questions.length > questionIndex) { return; }
    if(onlySkipped) { return onSubmit(); }

    setOnlySkipped(true);

    const skippedIndexes = questions
      .map(({answer}, index) => ({answer, index}))
      .filter(({answer}) => !answer.answerId)
      .map(({index}) => index);

    if(skippedIndexes.length === 0) { return onSubmit(); }
    if(window.confirm(translate("cognitive_confirm_retry"))) {
      setSkipped(skippedIndexes);
      setQuestionIndex(skippedIndexes[0]);
    } else {
      return onSubmit();
    }
  }, [questions, questionIndex]);
  const question = questions[questionIndex];
  const nextQuestion = questions[questionIndex + 1];

  if(!assessment) { return null; }
  if(assessment.completed) { return null; }
  if(questionIndex === null) {
    return (
      <Instructions
        captureLearningDisability={options.captureLearningDisability}
        initialLearningDisability={options.initialLearningDisability}
        onStart={onStart}
        options={options}
        translate={translate}
      />
    );
  }
  if(!question || submitting.current) { return <Loading />; }

  const index = skipped ? skipped.indexOf(questionIndex) : questionIndex;
  const total = skipped ? skipped.length : questions.length;
  const progress = (100.0 * (index + 1)) / total;

  return (
    <div className={style.container}>
      <div className={style.statusContainer}>
        {!options.disableTimeLimit && (
          <Timer
            onFinish={onSubmit}
            startTime={startTime}
            timeAllowed={
              disability
                ? options.specialTimeLimit || assessment.specialAllottedTime
                : options.timeLimit || assessment.allottedTime
            }
          />
        )}
        <div className={style.status}>
          {skipped && <span>{translate("cognitive_skipped_questions")} </span>}
          <span>{index + 1} / {total}</span>
        </div>
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
      </div>
      <Slide
        onSelect={onSelect}
        question={question}
        translate={translate}
      />
      {nextQuestion && (
        <Slide
          className={style.hide}
          onSelect={onSelect}
          question={nextQuestion}
          translate={translate}
        />
      )}
    </div>
  );
}
