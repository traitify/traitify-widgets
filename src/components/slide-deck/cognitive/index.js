/* eslint-disable no-alert */
import PropTypes from "prop-types";
import {useEffect, useRef, useState} from "react";
import Loading from "components/loading";
import {update as updateQuery} from "lib/graphql/queries/cognitive";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import Instructions from "./instructions";
import Slide from "./slide";
import Timer from "./timer";
import {useQuestionsLoader} from "./helpers";
import style from "./style.scss";

function Cognitive(props) {
  const {
    assessment,
    cache,
    getCognitiveAssessment,
    getOption,
    isReady,
    traitify,
    translate,
    ui
  } = props;
  const disableTimeLimit = getOption("slideDeck", "disableTimeLimit");
  const [initialQuestions, setInitialQuestions] = useState([]);
  const {dispatch, questions} = useQuestionsLoader(initialQuestions);
  const [disability, setDisability] = useState(false);
  const [onlySkipped, setOnlySkipped] = useState(false);
  const [questionIndex, setQuestionIndex] = useState(null);
  const [skipped, setSkipped] = useState(null);
  const [startTime, setStartTime] = useState(null);
  const [submitAttempts, setSubmitAttempts] = useState(0);
  const submitting = useRef(false);
  const state = {
    disability,
    dispatch,
    initialQuestions,
    questions,
    onlySkipped,
    questionIndex,
    skipped,
    startTime
  };
  const onSelect = (answer) => {
    dispatch({answer, questionIndex, type: "response"});
    if(!onlySkipped) { return setQuestionIndex(questionIndex + 1); }

    const newQuestionIndex = questions
      .findIndex((question, index) => index > questionIndex && !question.answer.answerId);
    setQuestionIndex(newQuestionIndex === -1 ? questions.length + 1 : newQuestionIndex);
  };
  const onStart = (options) => {
    if(options.disability) { setDisability(true); }

    setQuestionIndex(0);
    setStartTime(Date.now());
  };
  const onSubmit = () => {
    if(isReady("results")) { return; }
    if(submitAttempts > 3) { return; }
    if(submitting.current) { return; }

    submitting.current = true;

    const answers = questions.map((question) => (
      question.answer ? ({
        answerId: question.answer.answerId,
        questionId: question.id,
        skipped: question.answer.skipped || false,
        timeTaken: question.answer.timeTaken < 1 ? 1 : question.answer.timeTaken
      }) : ({
        questionId: question.id,
        skipped: true, // Currently required by API
        timeTaken: 1 // Currently required by API
      })
    ));
    const query = updateQuery({
      params: {
        answers,
        learningDisability: disability,
        overallTimeTaken: parseInt((Date.now() - startTime) / 1000, 10),
        testId: assessment.id
      }
    });

    traitify.post("/cognitive-tests/graphql", query).then(({data, errors}) => {
      if(!errors && data.completeCognitiveTest.success) {
        ui.trigger("SlideDeck.finished", {props, state});
        // If actual results are shown, use this, but for now it won't work because the API is async
        // getCognitiveAssessment({force: true});
        const key = `${assessment.localeKey.toLowerCase()}.cognitive-assessment.${assessment.id}`;
        cache.set(key, {...assessment, completed: true});
        getCognitiveAssessment();
      } else {
        console.warn(errors || data); // eslint-disable-line no-console
        submitting.current = false;
        setSubmitAttempts(submitAttempts + 1);
      }
    });
  };

  useDidMount(() => { ui.trigger("SlideDeck.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("SlideDeck.updated", {props, state}); });
  useDidUpdate(() => { onSubmit(); }, [submitAttempts]);
  useEffect(() => {
    if(!assessment) { return; }

    const cachedData = cache.get(`cognitive.slide-deck.${assessment.id}`);
    if(cachedData) {
      setDisability(cachedData.disability);
      setOnlySkipped(cachedData.onlySkipped);
      setSkipped(cachedData.skipped);

      setQuestionIndex(cachedData.questionIndex);
      setInitialQuestions(cachedData.questions);
      setStartTime(cachedData.startTime);
    } else {
      setInitialQuestions(assessment.questions);
    }
  }, [
    assessment && assessment.questions,
    assessment && assessment.localeKey
  ]);

  useEffect(() => {
    if(!assessment) { return; }
    if(questions.length === 0) { return; }

    cache.set(`cognitive.slide-deck.${assessment.id}`, {
      disability,
      onlySkipped,
      questions,
      questionIndex,
      skipped,
      startTime
    });
  }, [disability, onlySkipped, questions, questionIndex, skipped, startTime]);

  useEffect(() => {
    if(questions.length === 0) { return; }
    if(questions.length > questionIndex) { return; }
    if(onlySkipped) { return onSubmit(); }

    setOnlySkipped(true);

    const skippedIndexes = questions
      .map(({answer}, index) => ({answer, index}))
      .filter(({answer}) => !answer.answerId)
      .map(({index}) => index);

    if(skippedIndexes.length === 0) { return onSubmit(); }
    if(window.confirm(translate("cognitive_confirm_retry"))) {
      setSkipped(skippedIndexes);
      setQuestionIndex(skippedIndexes[0]);
    } else {
      return onSubmit();
    }
  }, [questions, questionIndex]);
  const question = questions[questionIndex];
  const nextQuestion = questions[questionIndex + 1];

  if(isReady("results")) { return null; }
  if(questionIndex === null) {
    return (
      <Instructions
        captureLearningDisability={getOption("slideDeck", "captureLearningDisability")}
        initialLearningDisability={getOption("slideDeck", "initialLearningDisability")}
        onStart={onStart}
        options={getOption("slideDeck") || {}}
        translate={translate}
      />
    );
  }
  if(!question || submitting.current) { return <Loading />; }

  const index = skipped ? skipped.indexOf(questionIndex) : questionIndex;
  const total = skipped ? skipped.length : questions.length;
  const progress = (100.0 * (index + 1)) / total;

  return (
    <div className={style.container}>
      <div className={style.statusContainer}>
        {!disableTimeLimit && (
          <Timer
            onFinish={onSubmit}
            startTime={startTime}
            timeAllowed={
              disability
                ? getOption("slideDeck", "specialTimeLimit") || assessment.specialAllottedTime
                : getOption("slideDeck", "timeLimit") || assessment.allottedTime
            }
          />
        )}
        <div className={style.status}>
          {skipped && <span>{translate("cognitive_skipped_questions")} </span>}
          <span>{index + 1} / {total}</span>
        </div>
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
      </div>
      <Slide
        className={style.slideContainer}
        onSelect={onSelect}
        question={question}
        translate={translate}
      />
      {nextQuestion && (
        <Slide
          className={style.hide}
          onSelect={onSelect}
          question={nextQuestion}
          translate={translate}
        />
      )}
    </div>
  );
}

Cognitive.defaultProps = {assessment: null};
Cognitive.propTypes = {
  assessment: PropTypes.shape({
    allottedTime: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
    localeKey: PropTypes.string.isRequired,
    questions: PropTypes.arrayOf(
      PropTypes.shape({id: PropTypes.string.isRequired})
    ).isRequired,
    specialAllottedTime: PropTypes.number.isRequired
  }),
  cache: PropTypes.shape({
    get: PropTypes.func.isRequired,
    set: PropTypes.func.isRequired
  }).isRequired,
  getCognitiveAssessment: PropTypes.func.isRequired,
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  traitify: TraitifyPropTypes.traitify.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {Cognitive as Component};
export default withTraitify(Cognitive);
