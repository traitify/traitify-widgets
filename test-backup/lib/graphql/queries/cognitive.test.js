import {create, get, surveys, update} from "lib/graphql/queries/cognitive";

describe("Graphql", () => {
  describe("Queries", () => {
    describe("Cognitive", () => {
      describe("create", () => {
        const params = {localeKey: "en-US", surveyId: "abc"};

        it("creates mutation", () => {
          const query = create({params, fields: ["completed", "id", {questions: ["id"]}]});

          expect(query).toMatchSnapshot();
        });

        it("creates mutation with defaults", () => {
          const query = create({params});

          expect(query).toMatchSnapshot();
        });
      });

      describe("get", () => {
        const params = {localeKey: "en-US", testId: "abc"};

        it("creates query", () => {
          const query = get({params, fields: ["completed", "id", {questions: ["id"]}]});

          expect(query).toMatchSnapshot();
        });

        it("creates query with defaults", () => {
          const query = get({params});

          expect(query).toMatchSnapshot();
        });
      });

      describe("surveys", () => {
        it("creates query", () => {
          const query = surveys({fields: {edges: {node: ["id"]}}});

          expect(query).toMatchSnapshot();
        });

        it("creates query with defaults", () => {
          const query = surveys();

          expect(query).toMatchSnapshot();
        });
      });

      describe("update", () => {
        const params = {
          answers: [{answerId: "a-1", questionId: "q-1", timeTaken: 100}],
          overallTimeTaken: 2500,
          testId: "abc"
        };

        it("creates mutation", () => {
          const query = update({params, fields: ["message"]});

          expect(query).toMatchSnapshot();
        });

        it("creates mutation with defaults", () => {
          const query = update({params});

          expect(query).toMatchSnapshot();
        });
      });
    });
  });
});
