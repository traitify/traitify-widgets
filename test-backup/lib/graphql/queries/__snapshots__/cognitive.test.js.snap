// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphql Queries Cognitive create creates mutation 1`] = `
{
  "query": "
      mutation($localeKey: String, $surveyId: String!) {
        createCognitiveTest(localeKey: $localeKey, surveyId: $surveyId) {
          completed id questions { id }
        }
      }
    ",
  "variables": {
    "localeKey": "en-US",
    "surveyId": "abc",
  },
}
`;

exports[`Graphql Queries Cognitive create creates mutation with defaults 1`] = `
{
  "query": "
      mutation($localeKey: String, $surveyId: String!) {
        createCognitiveTest(localeKey: $localeKey, surveyId: $surveyId) {
          allottedTime completed id learningDisability localeKey name percentileScore rawScore specialAllottedTime surveyId surveyKey questions { id answer { answerId skipped timeTaken } questionImage { id url } responses { id image { id url } } }
        }
      }
    ",
  "variables": {
    "localeKey": "en-US",
    "surveyId": "abc",
  },
}
`;

exports[`Graphql Queries Cognitive get creates query 1`] = `
{
  "query": "
      query cognitiveTest($localeKey: String, $testId: String!) {
        cognitiveTest(localeKey: $localeKey, testId: $testId) {
          completed id questions { id }
        }
      }
    ",
  "variables": {
    "localeKey": "en-US",
    "testId": "abc",
  },
}
`;

exports[`Graphql Queries Cognitive get creates query with defaults 1`] = `
{
  "query": "
      query cognitiveTest($localeKey: String, $testId: String!) {
        cognitiveTest(localeKey: $localeKey, testId: $testId) {
          allottedTime completed id learningDisability localeKey name percentileScore rawScore specialAllottedTime surveyId surveyKey questions { id answer { answerId skipped timeTaken } questionImage { id url } responses { id image { id url } } }
        }
      }
    ",
  "variables": {
    "localeKey": "en-US",
    "testId": "abc",
  },
}
`;

exports[`Graphql Queries Cognitive surveys creates query 1`] = `
{
  "query": "
      query($first: Int = 50) {
        cognitiveSurveys(first: $first) {
          edges { node { id } }
        }
      }
    ",
  "variables": undefined,
}
`;

exports[`Graphql Queries Cognitive surveys creates query with defaults 1`] = `
{
  "query": "
      query($first: Int = 50) {
        cognitiveSurveys(first: $first) {
          edges { node { id key name } }
        }
      }
    ",
  "variables": undefined,
}
`;

exports[`Graphql Queries Cognitive update creates mutation 1`] = `
{
  "query": "
      mutation(
        $answers: [QuestionAnswer]!,
        $learningDisability: Boolean,
        $overallTimeTaken: Int!,
        $testId: String!
      ) {
        completeCognitiveTest(
          answers: $answers,
          learningDisability: $learningDisability,
          overallTimeTaken: $overallTimeTaken,
          testId: $testId
        ) {
          message
        }
      }
    ",
  "variables": {
    "answers": [
      {
        "answerId": "a-1",
        "questionId": "q-1",
        "timeTaken": 100,
      },
    ],
    "overallTimeTaken": 2500,
    "testId": "abc",
  },
}
`;

exports[`Graphql Queries Cognitive update creates mutation with defaults 1`] = `
{
  "query": "
      mutation(
        $answers: [QuestionAnswer]!,
        $learningDisability: Boolean,
        $overallTimeTaken: Int!,
        $testId: String!
      ) {
        completeCognitiveTest(
          answers: $answers,
          learningDisability: $learningDisability,
          overallTimeTaken: $overallTimeTaken,
          testId: $testId
        ) {
          message success
        }
      }
    ",
  "variables": {
    "answers": [
      {
        "answerId": "a-1",
        "questionId": "q-1",
        "timeTaken": 100,
      },
    ],
    "overallTimeTaken": 2500,
    "testId": "abc",
  },
}
`;
